src/stats/mod.rs

```diff
@@ -0,0 +1,402 @@
+// for now, we use some of the stats defined in the protocol crates
+pub use protocol_memcache::*;
+
+use ahash::HashMap;
+use metriken::Lazy;
+use paste::paste;
+use std::concat;
+
+type Duration = clocksource::Duration<clocksource::Nanoseconds<u64>>;
+
+pub static PERCENTILES: &[(&str, f64)] = &[
+    ("p25", 25.0),
+    ("p50", 50.0),
+    ("p75", 75.0),
+    ("p90", 90.0),
+    ("p99", 99.0),
+    ("p999", 99.9),
+    ("p9999", 99.99),
+];
+
+pub struct Snapshot {
+    pub prev: HashMap<Stat, u64>,
```

Comment:
I think snapshot should have a timestamp, then you can implement a `rate` method that can enumerate members of the Stat and calculate rates in a terse way.

Reply:
#14 opened for this